### ISSUE TITLE
Counting distinct and new users within fct_ga4__pages is incorrect outside of hour-grain

### ISSUE THREAD

**Issue by adamribaudo-velir** [Velir/dbt-ga4#125]
We have the following select statement in `fct_ga4__pages`:

```
        extract( hour from (select  timestamp_micros(event_timestamp))) as hour,
        page_key,
        ...
        count(distinct user_pseudo_id ) as distinct_user_pseudo_ids,
        sum( if(ga_session_number = 1,1,0)) as new_user_pseudo_ids,
```

Won't `distinct_user_pseudo_ids` be a count of distinct users within the hour? Summing across hours will not longer yield a distinct count. Similar issue with the `new_user_pseudo_ids` as summing across the hours will double-count these users. 

I'd love to open a wider discussion about how to handle distinct counts across multiple grains. What are the best practices? If you pre-aggregate, you can't change the grain.

**Comment by dgitis** [comment]
You are correct.

I honestly thought that I had removed hour from this model a couple of weeks after I first implemented it. I'm for removing hour.

I'm not even sure that we want pages models in the package at all any more. It seems like a premature optimization. Users should create `fct_ga4__event_page_view` models and then, based on the individual reporting requirements by site, they may want to write tables further optimized for reporting. Based on my experience with the package in the last 9+ months on about 10 sites, those requirements end up being very client-dependent.

Hourly traffic reporting is a good example of a client-dependent requirement but also, virtually any custom field added to a `page_view` tag ends up requiring that I override `fct_ga4__pages` or `dim_ga4__pages`.

For now, I'm happy with removing `hour` as an expedient solution. There will still be session splitting at the end of the day, but the impact of that is much smaller.

**Comment by adamribaudo-velir** [comment]
What I uncovered as a cost-effective way to produce distinct user counts across variable time ranges is to use Looker Studio parameters: https://support.google.com/looker-studio/answer/10588439?hl=en

This allows the BI tool to set a parameter in the `where` clause that isn't in the `select` clause. So:

```
select count(distinct user_pseudo_id) as Users from dbt_dev_aribaudo.stg_ga4__event_session_start
where event_date_dt >= PARSE_DATE('%Y%m%d', @DS_START_DATE) and event_date_dt <= PARSE_DATE('%Y%m%d', @DS_END_DATE)
```

This returns an accurate count of distinct users from any time range without needing to pre-aggregate anything or scan unnecessary partitions.

**Comment by dgitis** [comment]
We could, alternatively, add a View after the `fct_ga4__pages`  model that does the cross-partition aggregation calculations without being dependent on a tool like Looker Studio.

```
select
  event_date_dt,
  hour,
  page_location,
  page_key,
  page_title,
  sum(page_views) as page_views,  -- cross-partition page_views,
  count(distinct user_pseudo_id) as distinct_user_pseudo_ids, -- cross-partition users
  ...
from {{ref('fct_ga4__pages`)}}
group by 1,2,3,4,5,6
```
Unless I'm mistaken, this cross-partition aggregation View would hit the partitions as long as an event date is specified and we don't use any window functions in the View.

This would result in data being cut off at the beginning and the end of the partition range, but not in between. This is not a problem in the pages model. It would be an issue if we were to apply this pattern to sessions which is something we might consider as a replacement to the currently un-partitioned Table.

**Comment by dgitis** [comment]
Resolved by #127 . Hour grain has been removed
